### PR TITLE
Fixed bug: tree in extension does not show icon

### PR DESCRIPTION
**Pull request by geleems**
Fixed bug: tree in extension does not show icon

### CODE REVIEW

**Comment by geleems** [comment]
Icon of tree node in extension is not shown because of ADS code update.
Previously type of icon had been in `string` type, and was changed to `UriComponent` or `Uri` type.
Now the URI is shown as `[object Object]` instead of icon path. -->

image

File: src/vs/workbench/parts/scm/electron-browser/scmViewlet.ts [hunk 1]
```diff
@@ -549,7 +549,7 @@ class ResourceRenderer implements IListRenderer<ISCMResource, ResourceTemplate>
 
 		if (icon) {
 			template.decorationIcon.style.display = '';
-			template.decorationIcon.style.backgroundImage = `url('${icon}')`;
+			template.decorationIcon.style.backgroundImage = `url('${icon.toString(true)}')`;
```
Comment (kevcunnane) [inline src/vs/workbench/parts/scm/electron-browser/scmViewlet.ts]: Please undo this change - it's coming from VSCode, and I verified this is the exact same in their codebase. It looks like they're using a function to read the URL value here

File: src/sql/parts/modelComponents/tree/treeComponentRenderer.ts [hunk 2]
```diff
@@ -144,7 +144,7 @@ export class TreeComponentRenderer extends Disposable implements IRenderer {
 	 */
 	public renderElement(tree: ITree, element: ITreeComponentItem, templateId: string, templateData: TreeDataTemplate): void {
 		const icon = this.themeService.getTheme().type === LIGHT ? element.icon : element.iconDark;
-		templateData.icon.style.backgroundImage = icon ? `url('${icon}')` : '';
+		templateData.icon.style.backgroundImage = icon ? `url('${icon.path}')` : '';
```
Comment (kevcunnane) [inline src/sql/parts/modelComponents/tree/treeComponentRenderer.ts]: Let's be consistent with how this is used elsewhere in the codebase to avoid errors. Per src/vs/workbench/browser/parts/views/customView.ts, we need to revive the URL and then do a toString(true) on it.
```ts
		const iconUrl = icon ? URI.revive(icon) : null;
...
		templateData.icon.style.backgroundImage = iconUrl ? `url('${iconUrl.toString(true)}')` : '';

```

File: src/vs/workbench/parts/scm/electron-browser/scmViewlet.ts (hunk 1 above)
Comment (geleems) [reply src/vs/workbench/parts/scm/electron-browser/scmViewlet.ts]: the icon is in `URI` type, and it is class. 
This following code will only return `url('[object Object]')` string, and there is no way to parse it back to original `URI` object once it is converted to `'[object Object]'` string.
```
`url('${icon}')`
```

Comment (kevcunnane) [reply src/vs/workbench/parts/scm/electron-browser/scmViewlet.ts]: Again, this code is in VSCode shipping right now. Are you going to tag with a SQL Carbon Edit and if so why? What use case does this fix / unblock?
- If this needs to be fixed and has visible user impact, we need to fix upstream. Next merge will lose your change anyhow.

Comment (kevcunnane) [reply src/vs/workbench/parts/scm/electron-browser/scmViewlet.ts]: I'd suggest the fix here is a PR in VSCode codebase, happy to see you do that but this isn't the right place to fix...

File: src/sql/parts/modelComponents/tree/treeComponentRenderer.ts (hunk 2 above)
Comment (geleems) [reply src/sql/parts/modelComponents/tree/treeComponentRenderer.ts]: `URI.revive(icon)` returns `URI` object by the way. Fixed.

File: src/vs/workbench/parts/scm/electron-browser/scmViewlet.ts (hunk 1 above)
Comment (geleems) [reply src/vs/workbench/parts/scm/electron-browser/scmViewlet.ts]: rolled back.
